tests/api/test_product_minimal_variant_price.py

```diff
@@ -0,0 +1,606 @@
+    )
+    content = get_graphql_content(response)
+    data = content["data"]["productVariantDelete"]
+    assert data["errors"] == []
+
+    mock_update_product_minimal_variant_price_task.delay.assert_called_once_with(
+        product.pk
+    )
+
+
+def test_CategoryDelete_also_deletes_products(
+    staff_api_client, product, category, permission_manage_products
+):
+    # Making sure the products associated with the category are deleted otherwise
+    # we need to update their "minimal_variant_price" if the category was on sale.
+    category.products.add(product)
+    query = """
+        mutation CategoryDelete($id: ID!) {
+            categoryDelete(id: $id) {
+                category {
+                    name
+                }
+                errors {
+                    field
+                    message
+                }
+            }
+        }
+    """
+    variables = {"id": to_global_id("Category", category.pk)}
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_products]
+    )
+    assert response.status_code == 200
+
+    content = get_graphql_content(response)
+    data = content["data"]["categoryDelete"]
+    assert data["errors"] == []
+    with pytest.raises(product._meta.model.DoesNotExist):
+        product.refresh_from_db()
+
+
+@patch(
+    "saleor.graphql.product.mutations.products"
+    ".update_products_minimal_variant_prices_of_catalogues_task"
+)
+def test_CollectionAddProducts_updates_minimal_variant_price(
+    mock_update_minimal_variant_prices_task,
+    staff_api_client,
+    sale,
+    collection,
+    product_list,
+    permission_manage_products,
+):
+    sale.collections.add(collection)
+    assert collection.products.count() == 0
+    query = """
+        mutation CollectionAddProducts($id: ID!, $products: [ID]!) {
+            collectionAddProducts(collectionId: $id, products: $products) {
+                collection {
+                    products {
+                        totalCount
+                    }
+                }
+                errors {
+                    field
+                    message
+                }
+            }
+        }
+    """
+    collection_id = to_global_id("Collection", collection.id)
+    product_ids = [to_global_id("Product", product.pk) for product in product_list]
+    variables = {"id": collection_id, "products": product_ids}
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_products]
+    )
+    content = get_graphql_content(response)
+    data = content["data"]["collectionAddProducts"]
+    assert data["errors"] == []
+
+    mock_update_minimal_variant_prices_task.delay.assert_called_once_with(
+        product_ids=[p.pk for p in product_list]
+    )
+
+
+@patch(
+    "saleor.graphql.product.mutations"
+    ".products.update_products_minimal_variant_prices_of_catalogues_task"
+)
+def test_CollectionRemoveProducts_updates_minimal_variant_price(
```

Comment:
I don't think that we have convention with camel case for API classes.
`test_collection_remove_products_updates_minimal_variant_price`

Reply:
Test names should tell you what it's about and in these ones I'm testing particular GraphQL endpoints that class names I'm using here. This way the test is more readable. You can straight away know what part of the test's name is the tested part and which one the expected behavior.

Reply:
I agree with @korycins here - changes that we add should respect conventions used in the codebase in general. From the name `test_collection_remove_products_updates_minimal_variant_price` you'll also know which operation we're testing, and if that's not enough, look at the query in the test.

Reply:
I've renamed the test to respect the conventions.